### ISSUE TITLE
Block specific items from being auto-imported

### ISSUE THREAD

**Issue by balthild** [rust-lang/rust-analyzer#17164]
There are some crate providing items with same names as well-known items from `std`. If the item from the crate is not an alias of the one from `std`, importing it causes errors. Most of the time, the errors are confusing, as it is really unusual, unless the crates' user has encountered it before.

For example, `anyhow::Ok` conflicts with `std::result::Result::Ok()`, and unlike the `std` one as an enum variant, it is a function. If I type `ok` in the editor and adopt the first candidate without checking the completion list carefully, `anyhow::Ok` will be imported, breaking other `Ok`s (e.g. the `Ok` in match patterns).

![image](https://github.com/rust-lang/rust-analyzer/assets/2662758/65aed62e-9d64-42b6-8236-565883961937)

![image](https://github.com/rust-lang/rust-analyzer/assets/2662758/be1299b8-7128-4986-9a16-80443c8a954a)

In my opinion, most of the time when using auto-import, well-known items like `Ok` should always be presented unless the unusual ones are specified purposely. It is not ergonomic to require developers to check such items carefully every times they use auto-import.

However, identifying which item is "well-known" seems to exceed the capabilities `rust-analyzer` should have. It is also not always possible to persuade the crates' maintainer to remove such footguns. As an alternative, `rust-analyzer` could be able to be configured to block specified items being auto-imported.

Related: https://github.com/dtolnay/anyhow/issues/201

**Comment by Veykril** [comment]
Yes, we should filter out auto-importing completions if they shadow something that's in scope (that satisfies the "well-known" concept you are speaking of, since `Ok` is in the prelude, and as such already in scope).
